### PR TITLE
Dịch "backprop_vn" - Phần 5

### LINKED ISSUE #968: Dịch "backprop_vn" - Phần 5
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_multilayer-perceptrons/backprop_vn.md#L270-L330

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by quangnhat185**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #968 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 1]
```diff
@@ -295,7 +295,13 @@ We compute tensors as gradients and need to retain all the intermediate variable
 Another reason is that we typically train with minibatches containing more than one variable, thus more intermediate activations need to be stored.
 -->
 
-*dịch đoạn phía trên*
+Khi huấn luyện các mạng nơ-ron, lan truyền xuôi và lan truyền ngược phụ thuộc lẫn nhau. 
+Cụ thể, với lan truyền xuôi, đồ thị tính toán đi theo hướng các phụ thuộc và tính tất cả các biến trên đường đi của nó. 
+Những biến đó sau đó được sử dụng trong lan truyền ngược khi chiều của đồ thị tính toán đi ngược lại.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
Những biến này sau đó được sử dụng trong lan truyền ngược khi thứ tự tính toán trên đồ thị bị đảo ngược lại. 
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 2]
```diff
@@ -295,7 +295,13 @@ We compute tensors as gradients and need to retain all the intermediate variable
 Another reason is that we typically train with minibatches containing more than one variable, thus more intermediate activations need to be stored.
 -->
 
-*dịch đoạn phía trên*
+Khi huấn luyện các mạng nơ-ron, lan truyền xuôi và lan truyền ngược phụ thuộc lẫn nhau. 
+Cụ thể, với lan truyền xuôi, đồ thị tính toán đi theo hướng các phụ thuộc và tính tất cả các biến trên đường đi của nó. 
+Những biến đó sau đó được sử dụng trong lan truyền ngược khi chiều của đồ thị tính toán đi ngược lại. 
+Hệ quả đó là ta cần lưu trữ các giá trị trung gian cho đến khi lan truyền ngược hoàn tất. 
+Đây cũng chính là một trong những lý do khiến lan truyền ngược đòi hỏi nhiều bộ nhớ hơn đáng kể so với thuật toán dự đoán đơn giản.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
Đây cũng chính là một trong những lý do khiến lan truyền ngược đòi hỏi nhiều bộ nhớ hơn đáng kể so với khi chỉ cần đưa ra dự đoán.  
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 3]
```diff
@@ -295,7 +295,13 @@ We compute tensors as gradients and need to retain all the intermediate variable
...
 -->
 
-*dịch đoạn phía trên*
+Khi huấn luyện các mạng nơ-ron, lan truyền xuôi và lan truyền ngược phụ thuộc lẫn nhau. 
+Cụ thể, với lan truyền xuôi, đồ thị tính toán đi theo hướng các phụ thuộc và tính tất cả các biến trên đường đi của nó. 
+Những biến đó sau đó được sử dụng trong lan truyền ngược khi chiều của đồ thị tính toán đi ngược lại. 
+Hệ quả đó là ta cần lưu trữ các giá trị trung gian cho đến khi lan truyền ngược hoàn tất. 
+Đây cũng chính là một trong những lý do khiến lan truyền ngược đòi hỏi nhiều bộ nhớ hơn đáng kể so với thuật toán dự đoán đơn giản.  
+Các tensor được tính dưới dạng gradient và các biến trung gian cần được lưu trữ để sử dụng trong quy tắc dây chuyền. 
+Một lý do nữa đó là chúng ta thường huấn luyện các minibatch chứa nhiều hơn một biến, do đó nhiều kích hoạt trung gian cần được lưu trữ hơn.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
Một lý do nữa đó là chúng ta thường huấn luyện các minibatch chứa nhiều hơn một biến, do đó nhiều giá trị kích hoạt trung gian cần được lưu trữ hơn. 
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 4]
```diff
@@ -310,7 +316,10 @@ Another reason is that we typically train with minibatches containing more than
 * Training requires significantly more memory and storage.
 -->
 
-*dịch đoạn phía trên*
+* Lan truyền xuôi tính và lưu trữ lần lượt các biến trung gian trong đồ thị tính toán định nghĩa bởi mạng nơ-ron.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
* Lan truyền xuôi lần lượt tính và lưu trữ các biến trung gian trong đồ thị tính toán định nghĩa bởi mạng nơ-ron. 
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 5]
```diff
@@ -310,7 +316,10 @@ Another reason is that we typically train with minibatches containing more than
 * Training requires significantly more memory and storage.
 -->
 
-*dịch đoạn phía trên*
+* Lan truyền xuôi tính và lưu trữ lần lượt các biến trung gian trong đồ thị tính toán định nghĩa bởi mạng nơ-ron. 
+* Lan truyền ngược tính và lưu trữ lần lượt các gradient của biến trung gian và tham số trong mạng nơ-ron theo chiều ngược lại.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
* Lan truyền ngược lần lượt tính và lưu trữ các gradient của biến trung gian và tham số trong mạng nơ-ron theo chiều ngược lại. 
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 6]
```diff
@@ -310,7 +316,10 @@ Another reason is that we typically train with minibatches containing more than
 * Training requires significantly more memory and storage.
 -->
 
-*dịch đoạn phía trên*
+* Lan truyền xuôi tính và lưu trữ lần lượt các biến trung gian trong đồ thị tính toán định nghĩa bởi mạng nơ-ron. 
+* Lan truyền ngược tính và lưu trữ lần lượt các gradient của biến trung gian và tham số trong mạng nơ-ron theo chiều ngược lại. 
+* Khi huấn luyện các mô hình học sâu, lan truyền xuôi và lan truyền ngược phụ thuộc lẫn nhau. 
+* Việc huấn luyện cần nhiều hơn đáng kể bộ nhớ và lưu trữ.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
* Việc huấn luyện cần nhiều bộ nhớ lưu trữ hơn đáng kể. 
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 7]
```diff
@@ -331,7 +340,15 @@ Another reason is that we typically train with minibatches containing more than
     * What are the advantages and disadvantages over training on a smaller minibatch?
 -->
 
-*dịch đoạn phía trên*
+1. Giả sử đầu vào $\mathbf{x}$ với hàm số vô hướng $f$ là ma trận $n \times m$. Chiều của gradient $f$ ứng với $\mathbf{x} là gì?
+2. Thêm vào một thiên kiến (bias) trong tầng ẩn của mô hình mô tả ở trên.
+  * Vẽ đồ thị tính toán tương ứng.
+  * Đạo hàm các phương trình lan truyền xuôi và lan truyền ngược.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
  * Tìm các phương trình cho quá trình lan truyền xuôi và lan truyền ngược.
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 8]
```diff
@@ -331,7 +340,15 @@ Another reason is that we typically train with minibatches containing more than
...
+3. Tính footprint bộ nhớ dùng để huấn luyện và tương tác với mô hình được mô tả ở chương này.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
3. Tính lượng bộ nhớ mà mô hình được mô tả ở chương này sử dụng lúc huấn luyện và lúc dự đoán.
```
Thường thì trong ML Inference == prediction nhé.

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 9]
```diff
@@ -343,8 +343,8 @@ Một lý do nữa đó là chúng ta thường huấn luyện các minibatch ch
 1. Giả sử đầu vào $\mathbf{x}$ với hàm số vô hướng $f$ là ma trận $n \times m$. Chiều của gradient $f$ ứng với $\mathbf{x} là gì?
 2. Thêm vào một thiên kiến (bias) trong tầng ẩn của mô hình mô tả ở trên.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
2. Thêm một hệ số điều chỉnh vào tầng ẩn của mô hình được mô tả ở trên.
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 10]
```diff
@@ -372,4 +389,4 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 5 -->
-*
+* Nguyễn Lê Quang Nhật
```
Comment (quangnhat185) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
* Nguyễn Lê Quang Nhật
* Lê Khắc Hồng Phúc
```
